code/modules/mob/living/simple_animal/vore/solargrub.dm

```diff
@@ -125,4 +122,70 @@ List of things solar grubs should be able to do:
 	. = ..()
 	if(. == 0 && !is_dead())
 		set_light(2.5, 1, COLOR_YELLOW)
-		return 1
\ No newline at end of file
+		return 1
+
+/mob/living/simple_animal/hostile/solargrubknight //more robust grub, doesn't like anyone.
+	name = "solargrub knight"
+	desc = "A not so friendly looking solargrub"
+	icon = 'icons/mob/vore64x32.dmi'
+	icon_state = "grubknight"
+	icon_living = "grubknight"
+	icon_dead = "grubknight-dead"
+
+	faction = "grubs"
+	maxHealth = 140 //BEEG KNIGHT
+	health = 140
+	move_to_delay = 4
+
+	melee_damage_lower = 5
+	melee_damage_upper = 10 //main damage from chems.
+
+	speak_chance = 1
+	emote_see = list("squelches", "squishes")
+
+	speed = 1
+	
+	meat_amount = 5
+	meat_type = /obj/item/weapon/reagent_containers/food/snacks/meat/grubmeat
+
+	response_help = "pokes"
+	response_disarm = "pushes"
+	response_harm = "roughly pushes"
+
+	min_oxy = 0
+	max_oxy = 0
+	min_tox = 0
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+	
+	var/poison_per_bite = 6
+	var/poison_type = "shockchem"
+	var/poison_chance = 65
+	var/emp_chance = 30
+	
+	vore_active = 1
+	vore_capacity = 1
+	vore_pounce_chance = 0
+	vore_default_mode = DM_ITEMWEAK
+
+/mob/living/simple_animal/hostile/solargrubknight/PunchTarget()
+	. = ..()
+	if(target_mob&& prob(emp_chance))
```

Comment:
This will let them EMP non-adjacent things, because ..() will return null if they aren't adjacent, but you're not checking it here, and the mob will still have target_mob even if it's moved away. Which is why you should override `do_punch` for your special attacks.